mono/mini/aot-compiler.c

```diff
@@ -11389,6 +11396,35 @@ add_preinit_got_slots (MonoAotCompile *acfg)
 	acfg->nshared_got_entries = acfg->got_offset;
 }
 
+static const char* interp_in_static_sigs[] = {
```

Comment:
How are those signatures computed?

Reply:
see commit message https://github.com/mono/mono/pull/6122/commits/b64faae88c30da83c842fc89edb8eb9f2a6174d9

Reply:
Why can't this be computed by the AOT compiler?

Reply:
These wrappers are needed for native2managed transitions. We would need to know all callsites from the embedder that are using the `compile_method ()` callback of the runtime and determine the used signatures. That's not possible with the AOT compiler.

Reply:
We should then require all embeders to annotate all managed functions they plan to call with a cattr we can look for during AOT. Similar to how the reverse pinvoke one works.

Reply:
For bitcode, we simply generate wrappers for all signatures in the app. With sharing, its not that big.

Reply:
@kumpera that would still require us to scan all assemblies ahead-of-time which we can't always do

Reply:
Absolutely but this setup is really fragile.
BCL + SDKs should not be manually curated.

Reply:
We are thinking about getting rid of this requirement to know ahead of time all the signatures, and do something similar for what we do for the `managed2native` transition today (aka. the `enter_icall_trampoline`.